src/main/java/de/bonndan/nivio/model/LandscapeItems.java

```diff
@@ -0,0 +1,152 @@
+package de.bonndan.nivio.model;
+
+import com.googlecode.cqengine.ConcurrentIndexedCollection;
+import com.googlecode.cqengine.IndexedCollection;
+import com.googlecode.cqengine.attribute.Attribute;
+import com.googlecode.cqengine.query.parser.sql.SQLParser;
+import com.googlecode.cqengine.resultset.ResultSet;
+import org.springframework.util.StringUtils;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static com.googlecode.cqengine.query.QueryFactory.attribute;
+import static de.bonndan.nivio.model.LandscapeItem.IDENTIFIER_VALIDATION;
+
+public class LandscapeItems {
+
+    private static final Attribute<Item, String> IDENTIFIER = attribute("identifier", Item::getIdentifier);
+    private static final Attribute<Item, String> NAME = attribute("name", Item::getName);
+
+    IndexedCollection<Item> index = new ConcurrentIndexedCollection<>();
+
+    /**
+     * @deprecated use only for testing
+     */
+    public static LandscapeItems of(List<Item> items) {
+        LandscapeItems landscapeItems = new LandscapeItems();
+        landscapeItems.setItems(new HashSet<>(items));
+        return landscapeItems;
+    }
+
+    public Stream<Item> stream() {
+        return index.stream();
+    }
+
+    public void setItems(Set<Item> items) {
+        index = new ConcurrentIndexedCollection<>();
+        index.addAll(items);
+    }
+
+    public void add(Item item) {
+        index.add(item);
+    }
```

Comment:
maybe Objects.requireNonNull for the items? (here and elsewhere)

Reply:
ends in NPE anyway, so can be done in develop branch